routers/web/web.go

```diff
@@ -108,7 +110,22 @@ func Routes(ctx gocontext.Context) *web.Route {
 	routes := web.NewRoute()
 
 	routes.Head("/", misc.DummyOK) // for health check - doesn't need to be passed through gzip handler
-	routes.RouteMethods("/assets/*", "GET, HEAD", CorsHandler(), public.AssetsHandlerFunc("/assets/"))
+
+	if setting.IsProd { // use assets from embed or filesystem
+		routes.RouteMethods("/assets/*", "GET, HEAD", CorsHandler(), public.AssetsHandlerFunc("/assets/"))
+	} else { // reverse-proxy asset requests to webpack-dev-server
+		// this environment variable is intentionally not loaded from ini config because webpack also needs
+		// to parse it from the environment
```

Comment:
```suggestion
		// this environment variable is intentionally not loaded from ini config
		// because webpack also needs to parse it from the environment
```